controllers/profiles/reconciler_dev.go

```diff
@@ -81,19 +83,31 @@ func newDevProfileReconciler(client client.Client, logger *logr.Logger) ProfileR
 func newDevelopmentObjectEnsurers(support *stateSupport) *devProfileObjectEnsurers {
 	return &devProfileObjectEnsurers{
 		deployment:          newObjectEnsurer(support.client, support.logger, defaultDeploymentCreator),
-		service:             newObjectEnsurer(support.client, support.logger, defaultServiceCreator),
+		service:             newObjectEnsurer(support.client, support.logger, devServiceCreator),
+		route:               newConditionedObjectEnsurer(support.client, support.logger, defaultRouteCreator, onlyForOpenShiftClusters),
```

Comment:
Instead, we can have `newDevelopmentObjectEnsurersForOpenShift`.

The ensurer then can be a `network` and we inject either the nodePort ensurer or the route one.